core/templates/dev/head/pages/exploration_player/SupplementalCardDirective.js

```diff
@@ -28,24 +28,33 @@ oppia.directive('supplementalCard', [
         '/pages/exploration_player/' +
         'supplemental_card_directive.html'),
       controller: [
-        '$scope', '$window', 'oppiaPlayerService',
+        '$scope', '$timeout', '$window', 'oppiaPlayerService',
         'playerPositionService', 'playerTranscriptService',
         'ExplorationObjectFactory', 'windowDimensionsService',
         'CONTENT_FOCUS_LABEL_PREFIX', 'TWO_CARD_THRESHOLD_PX',
         'EVENT_ACTIVE_CARD_CHANGED', 'CONTINUE_BUTTON_FOCUS_LABEL',
+        'WAIT_FOR_HINT_MSEC',
         function(
-          $scope, $window, oppiaPlayerService,
+          $scope, $timeout, $window, oppiaPlayerService,
           playerPositionService, playerTranscriptService,
           ExplorationObjectFactory, windowDimensionsService,
           CONTENT_FOCUS_LABEL_PREFIX, TWO_CARD_THRESHOLD_PX,
-          EVENT_ACTIVE_CARD_CHANGED, CONTINUE_BUTTON_FOCUS_LABEL) {
+          EVENT_ACTIVE_CARD_CHANGED, CONTINUE_BUTTON_FOCUS_LABEL,
+          WAIT_FOR_HINT_MSEC) {
           var updateActiveCard = function() {
             var index = playerPositionService.getActiveCardIndex();
             if (index === null) {
               return;
             }
             $scope.activeCard = playerTranscriptService.getCard(index);
             $scope.clearHelpCard();
+            $scope.isCurrentHintUsable = false;
```

Comment:
I think you almost want to pull this stuff into its own HintManagerService (in a separate file that gets included in the player view) -- something that privately maintains a currentHintIsUsable boolean, a numHintsConsumed integer, and a timeout, and that exposes functions like:
- isCurrentHintUsable()
- activateHintAfterTimeout(timeoutMsec)
- clearTimeout()
- areAllHintsExhausted()
- reset()

Mingling it with all the other stuff seems a bit complex. By pulling it out into a service you get some separation of concerns and you can test it.